starter_code/src/movies.js

```diff
@@ -1,20 +1,146 @@
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+var movies = require('../src/data');
+
+/** Iteration  1
+ Turn duration of the movies from hours to minutes
+ Break it down into several chunks */
+
+// Removes `h`
+function hoursToMin(hours) {
+  let minFromHours = hours.split('h');
+  return minFromHours[0] * 60;
+}
+
+// Removes `min`
+function minToMin(min) {
+  let minFromMin = min.split('min');
+  return Number(minFromMin[0]);
+}
+
+// Removes the chars `h` and `min` from the duration key
+function doMath(duration) {
+  let timePieces = duration.split(' ');
+
+  let minutes = timePieces.reduce((sum, piece) => {
+    if (piece.includes('h')) {
+      return sum + hoursToMin(piece);
+    }
+    return sum + minToMin(piece);
+  }, 0);
+  return minutes;
+}
+
+// Main function
+
+function turnHoursToMinutes(movies) {
+  var newMoviesArray = movies.map(function (movie) {
+    var newMovie = {};
+    newMovie.title = movie.title;
+    newMovie.year = movie.year;
+    newMovie.director = movie.director;
+    newMovie.duration = doMath(movie.duration);
+    newMovie.genre = movie.genre;
+    newMovie.rate = movie.rate;
+    return newMovie;
+  });
+  return newMoviesArray;
+}
+
+turnHoursToMinutes(movies);
+
+// ----------------------------------------------------------
+// Iteration  2
+// Get the average of all rates with 2 decimals
+// reduce method?
+// function ratesAverage(arr) {}
 
 
 // Get the average of all rates with 2 decimals 
+function ratesAverage(movies) {
+  var theSum = movies.reduce(function (sum, movie) {
+    return sum + Number(movie.rate);
+  }, 0);
+  var rateAvg = theSum / movies.length;
+  return Math.floor(rateAvg * 100) / 100
```

Comment:
for a rating you would want to use Math.round() not Math.floor() since you don't want to lower a rating that would be 4.7 to be a 4 rather than a 5 that it should be.